src/basic/log.h

```diff
@@ -367,3 +367,37 @@ int log_syntax_invalid_utf8_internal(
 #define DEBUG_LOGGING _unlikely_(log_get_max_level() >= LOG_DEBUG)
 
 void log_setup(void);
+
+struct log_context {
+        char *const *fields;
+        struct log_context *prev;
+};
+
+extern thread_local struct log_context *_log_context;
+
+static inline void _log_context_pop(struct log_context **c) {
+        if (!*c)
```

Comment:
`if (!c || !*c)`

Reply:
We don't do this check either in `DEFINE_TRIVIAL_CLEANUP_FUNC()` so I left it out here as well. I think it's guaranteed c itself won't be `NULL`?

Reply:
force of habit... doesn't cost much to check, and it's a bit friendlier to onlookers who might not know what is guaranteed and what is not - in fact if it's guaranteed, then an assert() to assert it would be a nice way to document it clearly